src/ethereum_test_tools/tests/test_vm.py

```diff
@@ -287,6 +284,49 @@
             b"\x60\x00\x60\x00\x60\x01\xF9",
             id="Op.EXTDELEGATECALL(address=1)",
         ),
+        pytest.param(
+            Om.MSTORE(b""),
+            b"",
+            id='Om.MSTORE(b"")',
+        ),
+        pytest.param(
+            Om.MSTORE(bytes(range(32))),
+            bytes(Op.MSTORE(0, bytes(range(32)))),
+            id="Om.MSTORE(bytes(range(32)))",
+        ),
+        pytest.param(
+            Om.MSTORE(bytes(range(64))),
+            bytes(Op.MSTORE(0, bytes(range(32))) + Op.MSTORE(32, bytes(range(32, 64)))),
+            id="Om.MSTORE(bytes(range(64)))",
+        ),
+        pytest.param(
+            Om.MSTORE(bytes(range(33))),
+            bytes(
+                Op.MSTORE(0, bytes(range(32)))
+                + Op.MLOAD(32)
```

Comment:
Why it so complicated here?

Shouldn't it be ?
```
mstore(0, bytes(range(32)))
mstore(32, 0x33...00)
```

Reply:
The reason is that it tries to preserve the memory outside of the range we are storing because we can only write 32-byte chunks, so we always assume there's already some information in the mstore range + 32 bytes, so we first read the existing memory range, mask it, then perform an OR with the new data, and finally store.